src/componnets/AddItem.jsx

```diff
@@ -0,0 +1,44 @@
+import React, { useState } from "react";
+import { IconButton } from "@mui/material";
+import AddBoxOutlinedIcon from "@mui/icons-material/AddBoxOutlined";
+import {
+  CategoryService,
+  ListService,
+  TodoService,
+} from "../services/httpService";
+import EditFields from "./EditFields";
+
+function AddItem({ type, itemId, setItem }) {
+  const [showFields, setShowFields] = useState(false);
+
+  const services = [
```

Comment:
Nah, its not the way to do it, I think this entire file has a problem, its too generic. 
we need to talk about it